fix(@clayui/css): LPD-55065 Update c-slideout and sidebar to adhere …

pat270 opened the pull request:
…to Lexicon Side Panel specs

https://liferay.atlassian.net/browse/LPD-55065
https://liferay.atlassian.net/browse/LPD-33184

pat270 commented:
@ethib137 I wasn't able to get something working for fixed header and footer without using `position: fixed`. I think we need to use position fixed and calculate the top bar heights if we want fixed header and footer.

![c-slideout](https://github.com/user-attachments/assets/666e9c2c-d68f-46ae-881a-51d1aa8575ff)

ethib137 commented:
Hey @pat270 , 

Thanks for the info on fixed vs absolute. I'll check with product and design and see what we have to do.

Regarding styling it looks like we are still missing a border for the header and footer. Additionally, it would be good to add a new storybook example that includes the additional elements in the header like we see [here](https://www.figma.com/design/vLPRyGECU7lFC86wsbd5Hy/design-5654-side-panel?node-id=47-17263&t=vAV4yIc4moFlZGCT-0). Thanks.

pat270 commented:
@ethib137 We might not need the position fixed. I misinterpreted the height.

> When we talk about “Component Height”, we are talking about the actual height of the component, not the total height of the browser.

I went ahead and added a min-height to the container since container queries for height don't work. See https://liferay.atlassian.net/browse/LPD-45006?focusedCommentId=2939028.

I also added a sticky prop to the Header and Footer so the dev can choose to have one or both fixed.

![sidePanelSticky](https://github.com/user-attachments/assets/3f055039-adf9-45e9-8726-5ceed8a741d8)

ethib137 commented on packages/clay-css/src/scss/cadmin/components/_sidebar.scss:
```diff
@@ -121,3 +145,23 @@
+	bottom: 0;
+	position: sticky;
+	z-index: $cadmin-zindex-sticky;
+}
```
Hey @pat270 we already have a sticky-top utility class. Why do we need the sidebar-header-sticky class? Also instead of sidebar-footer-sticky class, maybe copy the current pattern and add a sticky-bottom class instead. These would be usable in more situations.

ethib137 commented on packages/clay-core/src/side-panel/Header.tsx:
```diff
@@ -13,14 +14,32 @@ type Props = {
+			className={classnames(
+				'sidebar-header',
+				{
+					'sidebar-header-sticky': sticky,
```
Let's use the sticky-top utility class instead.

ethib137 commented on packages/clay-core/stories/SidePanel.stories.tsx:
```diff
@@ -94,14 +120,83 @@ export const Default = () => {
 				>
-					<SidePanel.Header>
-						<SidePanel.Title>Right Panel</SidePanel.Title>
+					<SidePanel.Header className="asdfadsf" sticky>
```
This looks like an unintended added className

pat270 replied:
oops thanks for getting this

ethib137 reviewed (changes requested):
Hey @pat270 this looking better. We are still going to need fixed positioning as an option though because we want to have independent scroll for the cms use case. I'm going to merge my PR so that you can rebase off of mine and make sure fixed and absolute positioning is working as expected. Thanks.
